src/site/content/en/blog/async-clipboard/index.md

```diff
@@ -0,0 +1,413 @@
+formats.
+{% endAside %}
+
+## Copy: writing data to the clipboard
+
+### writeText()
+
+To copy text to the clipboard call `writeText()`. Since this API is
+asynchronous, the `writeText()` function returns a Promise that resolves or
+rejects depending on whether the passed text is copied successfully:
+
+```js
+async function copyPageUrl() {
+  try {
+    await navigator.clipboard.writeText(location.href);
+    console.log('Page URL copied to clipboard');
+  } catch (err) {
+    console.error('Failed to copy: ', err);
+  }
+}
+```
+
+### write()
+
+Actually, `writeText()` is just a convenience method for the generic `write()`
+method, which also lets you copy images to the clipboard. Like `writeText()`, it
+is asynchronous and returns a Promise.
+
+To write an image to the clipboard, you need the image as a
+`[blob](https://developer.mozilla.org/en-US/docs/Web/API/blob)`. One way to do
+this is by requesting the image from a server using `fetch()`, then calling
+`[blob()](https://developer.mozilla.org/en-US/docs/Web/API/Body/blob)` on the
+response.
+
+Requesting an image from the server may not be desirable or possible for a
+variety of reasons. Fortunately, you can also draw the image to a canvas and
+call the canvas'
+`[toBlob()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob)`
+method.
+
+Next, pass an array of `ClipboardItem` objects as a parameter to the `write()`
+method. Currently you can only pass one image at a time, but we hope to add
+support for multiple images in the future. `ClipboardItem` takes an object with
+the MIME type of the image as the key and the blob as the value. For Blob
+objects obtained from fetch() or canvas.toBlob(), the `blob.type` property
+automatically contains the correct MIME type for an image.
+
+```js
+try {
+  const imgURL = '/images/generic/file.png';
+  const data = await fetch(imgURL);
+  const blob = await data.blob();
+  await navigator.clipboard.write([
+    new ClipboardItem({
+      [blob.type]: blob
+    })
+  ]);
+  console.log('Image copied.');
+} catch(err) {
+  console.error(err.name, err.message);
+}
+```
+
+### The copy event
+
+In the case where a user initiates a clipboard copy, non-textual data is
+provided as a blob for you. The
+[`copy` event](https://developer.mozilla.org/en-US/docs/Web/API/Document/copy_event)
+includes a `clipboardData` property with the items already in the right format,
+eliminating the need to manually create a blob. Call `preventDefault()` to
+prevent the default behavior in favor of your own logic, then copy contents to
+the clipboard. What's not covered in this example is how to fall back to earlier
+APIs when the Clipboard API isn't supported. I'll cover that under Feature
+detection, later in this article.
+
+```js
+document.addEventListener('copy', async (e) => {
+  e.preventDefault();
+  try {
+    let clipboardItems = [];
+    for (const item of e.clipboardData.items) {
+      if (!blog.type.startsWith('image/)') { continue; }
```

Comment:
```suggestion
      if (!blob.type.startsWith('image/)') { continue; }
```

Fun typo. Also `blob` comes out of nowhere. It's undefined.

Reply:
What do you think of wrapping this in something like:

```js
var canvas = document.getElementById('canvas');
canvas.toBlob(function(blob) {
  // 
});
```